velox/type/Variant.h

```diff
@@ -396,11 +396,12 @@ class variant {
     }
   }
 
-  std::string toJson(const TypePtr& type = nullptr) const;
+  std::string toJson(const TypePtr& type) const;
```

Comment:
@mbasmanova With this change, all `variant::toJson()` calls are invoked using a valid TyprPtr. The default `nullptr` can be removed. This will avoid additional checks and handling for `type == nullpr` inside toJson().

Reply:
That sounds good, but doesn't match the PR title / description. Let's make this change in a separate PR.

Reply:
It would be best to limit each PR to one fix.

Reply:
Fixed. Will raise another PR.

Reply:
@Krishna-Prasad-P-V let's move `toJson()` recursive fix separately by using 
```
  friend std::ostream& operator<<(std::ostream& stream, const variant& k) {
    const auto type = inferType();
    stream << k.toJson(type);
    return stream;
  }
```